pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -152,7 +152,10 @@ func (pl *PodTopologySpread) PreFilter(ctx context.Context, cycleState *framewor
 	s, err := pl.calPreFilterState(ctx, pod)
```

Comment:
nit for follow-up: in calPreFilterState() the logic of `allNodes, err := pl.sharedLister.NodeInfos().List()` can be moved to its most recent usage, so we save the cost of calling pl.sharedLister.NodeInfos().List() every time.

Reply:
/hold
I would just do it in this PR. We have time before the freeze.

Reply:
🆗

Reply:
I have moved its evaluation location.

Reply:
@alculquicondor Thanks for your review. PTAL